### ISSUE TITLE
Misleading ERROR log for missing statuses_url when using GITHUBPULL bug tracker

### ISSUE THREAD

**Issue by DLTKYugalGarg** [checkmarx-ts/checkmarx-cxflow-github-action#91]
## Description

When using `bug_tracker: GITHUBPULL` in GitHub Actions mode, CxFlow logs an ERROR message about missing `statuses_url` even though this field is not required for the GITHUBPULL bug tracker configuration.

## Environment

- **CxFlow Version**: 1.7.11 (via checkmarx-cxflow-github-action@v2.3)
- **Execution Mode**: GitHub Actions (CLI mode)
- **Bug Tracker**: GITHUBPULL
- **Platform**: GitHub Actions (ubuntu-latest)

## Current Behavior

The following ERROR appears in logs during scan execution:

```
2025-11-14 14:42:51.395 ERROR 7 --- [           main] c.c.f.s.GitHubService [ydG45hey] : 
statuses_url was not provided within the request object, which is required for 
blocking / unblocking pull requests
```

**Scan behavior**: Continues successfully and completes without issues

## Expected Behavior

The log message should be downgraded to WARN or removed entirely for GITHUBPULL bug tracker, since:

1. **GITHUBPULL** posts PR comments via `/repos/{owner}/{repo}/issues/{pr_number}/comments` API
2. **statuses_url** is only required for **GITHUB** bug tracker (commit status checks)
3. The scan completes successfully despite the ERROR log
4. GitHub Actions mode doesn't provide webhook payloads with statuses_url

## Why This is Confusing

The ERROR severity suggests a critical failure, but:
- Scan execution is not impacted
- PR comments are posted successfully
- Build breaking works correctly via `break_build: true`
- No functional issue exists

## Suggested Fix

**Option 1**: Conditional logging based on bug tracker type
```java
if (request.getStatusesUrl() == null) {
    if ("GITHUB".equals(bugTracker)) {
        log.error("statuses_url required for GITHUB bug tracker");
    } else if ("GITHUBPULL".equals(bugTracker)) {
        log.debug("statuses_url not needed for GITHUBPULL bug tracker");
    }
}
```

**Option 2**: Downgrade to WARN
```java
log.warn("statuses_url not provided - commit status checks unavailable");
```

## Reproduction Steps

1. Configure GitHub Action with GITHUBPULL bug tracker:
```yaml
- uses: checkmarx-ts/checkmarx-cxflow-github-action@v2.3
  with:
    bug_tracker: GITHUBPULL
    incremental: true
    # ... other standard params
```

2. Trigger on pull request event
3. Observe ERROR log during scan execution
4. Note scan completes successfully despite ERROR

## Impact

- **Severity**: Low (cosmetic issue)
- **User confusion**: Medium (ERROR suggests failure when none exists)
- **Functional impact**: None

## Additional Context

This appears to be in `GitHubService.java` where the code checks for `statuses_url` without considering the bug tracker mode. The check seems designed for webhook mode but runs in all execution modes.

## Workaround

Current workaround: Ignore the ERROR log message - scan completes successfully.

---

**Related Configuration**:
- Using GitHub Actions trigger (not webhook server mode)
- Break build enforcement via `break_build: true` parameter
- PR feedback via inline comments (not status checks)
